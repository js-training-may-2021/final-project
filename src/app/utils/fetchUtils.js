import axios from 'axios';
import _ from 'lodash';

const isDev = process.env.NODE_ENV === 'development';
const pathApi = isDev ? 'http://localhost:3000' : '/api/v1';
export const limit = 20;

const fetchPokemons = (page, query, attr, order) => axios.get(`${pathApi}/pokemons?q=${query}&_page=${page}&_limit=${limit}&_sort=${attr}&_order=${order}`);
const normalizedQuery = (query) => _.trimStart(query, '#');

export const mappingFetchPokemons = {
  idAsc: (page, query) => fetchPokemons(page, normalizedQuery(query), 'id', 'asc'),
  idDesc: (page, query) => fetchPokemons(page, normalizedQuery(query), 'id', 'desc'),
  nameAsc: (page, query) => fetchPokemons(page, normalizedQuery(query), 'name', 'asc'),
  nameDesc: (page, query) => fetchPokemons(page, normalizedQuery(query), 'name', 'desc'),
};

export const fetchPokemonById = (id) => axios.get(`${pathApi}/pokemons?id=${id}`);

export const fetchData = () => (
  axios({
    method: 'post',
    url: '/api/v1/data',
    data: { token: localStorage.token },
    timeout: 4000,
  })
);

export const fetchCatch = (pokemon) => axios({
  method: 'post',
  url: '/api/v1/catchPokemon',
  data: { ...pokemon, token: localStorage.token },
  timeout: 4000,
});

export const fetchLogin = (data) => axios({
  method: 'post',
  url: '/api/v1/login',
  data,
  timeout: 4000,
});
