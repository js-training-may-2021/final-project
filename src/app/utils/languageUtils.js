import { useTranslation } from 'react-i18next';

export const setDefaultLanguage = () => {
  const defaultLanguage = 'ru';

  if (localStorage.getItem('language') === null) {
    localStorage.setItem('language', defaultLanguage);
  }
};

export const getCurrentLanguage = () => localStorage.getItem('language');

export const toggleLanguage = () => {
  const currentLanguage = localStorage.getItem('language');
  const newLanguage = currentLanguage === 'ru' ? 'en' : 'ru';
  localStorage.setItem('language', newLanguage);

  return newLanguage;
};

export const loc = (text) => {
  const { t } = useTranslation();
  return t(text);
};
