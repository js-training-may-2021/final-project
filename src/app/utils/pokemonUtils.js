import _ from 'lodash';

export const isFound = ({ name, id }, searchText) => {
  const searchBy = (attr, text) => String(attr).includes(text);
  const normalizedText = _.trimStart(searchText.trim().toLowerCase(), '#');
  return searchBy(name, normalizedText) || searchBy(id, normalizedText);
};

export const getNextId = (currentId) => {
  const nextId = currentId + 1 === 803 ? 10001 : currentId + 1;
  return nextId === 10148 ? 1 : nextId;
};

export const getPreviousId = (currentId) => {
  const prevId = currentId - 1 === 0 ? 10147 : currentId - 1;
  return prevId === 10000 ? 802 : prevId;
};

export const mappingSortPokemons = {
  idAsc: (array) => _.orderBy(array, 'id', 'asc'),
  idDesc: (array) => _.orderBy(array, 'id', 'desc'),
  nameAsc: (array) => _.orderBy(array, 'name', 'asc'),
  nameDesc: (array) => _.orderBy(array, 'name', 'desc'),
};
