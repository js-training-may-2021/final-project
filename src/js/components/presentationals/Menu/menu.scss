.menu {
  display: flex;
  margin: 0;
  
  &__item {
    margin-right: 20px;
  }
}
