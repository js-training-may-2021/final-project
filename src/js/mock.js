export const pokemons = [
  {
    name: 'bulbasaur',
    id: 1,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'ivysaur',
    id: 2,
    isCaught: true,
    catchDate: '23 June 2021',
  },
  {
    name: 'venusaur',
    id: 3,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'charmander',
    id: 4,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'charmeleon',
    id: 5,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'charizard',
    id: 6,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'squirtle',
    id: 7,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'wartortle',
    id: 8,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'blastoise',
    id: 9,
    isCaught: true,
    catchDate: '23 June 2021'
  },
  {
    name: 'caterpie',
    id: 10,
    isCaught: false,
    catchDate: null,
  },
  {
    name: 'metapod',
    id: 11,
    isCaught: false,
    catchDate: null,
  }];  

export const uncaughtPokemon = pokemons[0];
export const caughtPokemon = pokemons[1];
