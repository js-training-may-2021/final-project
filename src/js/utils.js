import {createMemoryHistory} from 'history';
import configureStore from 'redux-mock-store';

export const POKEMONS_PER_PAGE = 10;

let currentPage = 1;

export const getCurrentPage = () => {
	return currentPage++;
};

export const AppRoute = {
  MAIN: '/',
  DETAIL: '/detail/:id',
  CAUGHT: '/caught'
};

const monthNames = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];

export const getCatchDate = () => {
  const date = new Date();

  const day = date.getDate();
  const month = monthNames[date.getMonth()];
  const year = date.getFullYear();

  return `${day} ${month} ${year}`
};

export const noop = () => {};
export const history = createMemoryHistory();
const mockStore = configureStore();
export const testStore = mockStore({});
