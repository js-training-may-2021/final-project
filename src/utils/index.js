export const firstLetterToUpperCase = (str) => {
  if (str === undefined) {
    return str;
  } else {
    return str
      .toLowerCase()
      .split(" ")
      .map((s) => s.charAt(0).toUpperCase() + s.substring(1))
      .join(" ");
  }
};

export const formatDate = (dateInput) => {
  return dateInput.toLocaleString("ru-RU");
};

export const padToThree = (number) => number <= 999 ? `00${number}`.slice(-3) : number;

export const handleErrors = (response) => {
  if (!response.ok) {
    throw Error(`${response.status}: ${response.statusText}`);
  }
  return response;
};
