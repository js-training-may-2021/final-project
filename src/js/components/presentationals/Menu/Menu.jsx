import React from 'react';
import './menu.scss';

const Menu = () => {
	return (
		<ul className="header__menu menu">
			<li className="menu__item">
				<a className="menu__link" href="./index.html">ALL</a>
			</li>
			<li className="menu__item">
				<a className="menu__link" href="#">CAUGHT</a>
			</li>
		</ul>
	);
};

export default Menu;
