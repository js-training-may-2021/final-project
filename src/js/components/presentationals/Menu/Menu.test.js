import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import Menu from './Menu.jsx';
import {AppRoute, history} from '../../../utils.js';

it(`Render Menu on Main page`, () => {
  const tree = renderer
    .create(
      <Router history={history}>
        <Menu />
      </Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Menu on Caught page`, () => {
  history.push(AppRoute.CAUGHT);

  const tree = renderer
    .create(
      <Router history={history}>
        <Menu />
      </Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render Menu on Detail page`, () => {
  history.push(AppRoute.DETAIL);
  
  const tree = renderer
    .create(
      <Router history={history}>
        <Menu />
      </Router>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
