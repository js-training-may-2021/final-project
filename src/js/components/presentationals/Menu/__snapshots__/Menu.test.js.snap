// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Menu on Caught page 1`] = `
<ul
  className="header__menu menu"
>
  <li
    className="menu__item"
  >
    <a
      aria-current={null}
      className="menu__link"
      href="/"
      onClick={[Function]}
    >
      ALL
    </a>
  </li>
  <li
    className="menu__item"
  >
    <a
      aria-current="page"
      className="menu__link  menu__link--selected"
      href="/caught"
      onClick={[Function]}
      style={Object {}}
    >
      CAUGHT
    </a>
  </li>
</ul>
`;

exports[`Render Menu on Detail page 1`] = `
<ul
  className="header__menu menu"
>
  <li
    className="menu__item"
  >
    <a
      aria-current={null}
      className="menu__link"
      href="/"
      onClick={[Function]}
    >
      ALL
    </a>
  </li>
  <li
    className="menu__item"
  >
    <a
      aria-current={null}
      className="menu__link"
      href="/caught"
      onClick={[Function]}
    >
      CAUGHT
    </a>
  </li>
</ul>
`;

exports[`Render Menu on Main page 1`] = `
<ul
  className="header__menu menu"
>
  <li
    className="menu__item"
  >
    <a
      aria-current="page"
      className="menu__link menu__link--selected"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      ALL
    </a>
  </li>
  <li
    className="menu__item"
  >
    <a
      aria-current={null}
      className="menu__link"
      href="/caught"
      onClick={[Function]}
    >
      CAUGHT
    </a>
  </li>
</ul>
`;
